Sources/MongoSwift/MongoClient.swift

```diff
@@ -4,6 +4,9 @@ import NIOConcurrencyHelpers
 
 /// Options to use when creating a `MongoClient`.
 public struct ClientOptions: CodingStrategyProvider, Decodable {
+    /// Specifies options this client should use when creating connection pools.
+    public var connectionPoolOptions: ConnectionPoolOptions?
```

Comment:
I have a slight preference for keeping the `ClientOptions` a flat structure, with a rationale that our connection string doesn't have hierarchies. In node's distant history it supported nested options like this, and it became quite unwieldy and confusing to users and developers alike. I don't really have a sense if that's the case here, and I see there's already precedence for this with our TLS options, but I figured I'd bring it up anyway. 

Maybe this is where to start: do you anticipate that a user will create a struct and pass it in?
```
var cmapOptions = ConnectionPoolOptions(maxPoolSize: 10)
var clientOptions = ClientOptions(conectionPoolOptions: cmapOptions, readConcern: .secondary)
var client = MongoClient(clientOptions)
```
That feels a little more cumbersome to me, but maybe this is a moot point because we expect people to mostly use the connection string.

Reply:
matching the structure of the connection string makes sense to me. I don't really recall why we went with the separate struct approach for TLS. possibly a leakage of internals since it made it easier to pass just the TLS options to the pool rather than all of the options.

I'm fine with switching to a flat structure for everything, will make the update